Include info on when to use stubbed objects (#130)
transcom/mymove-docs: docs/backend/testing/testing-best-practices.md

- Subtests must be able to run in isolation. You can verify this with Goland, which makes it easy to run individual subtests.
- Avoid `mock.anything`. Part of mocking is to ensure that the object that is mocked has been called with the right arguments. If a handler is supposed to call a service object with a specific model ID, then if you use `mock.anything` in the test and you introduce a bug in the handler where it passes in the wrong ID, the test will still pass. If you find it hard to avoid using `mock.anything`, it's most likely a sign the code under test is not designed properly.
- Always validate `Params` in handlers using `suite.NoError(params.Body.Validate(strfmt.Default))`

> - When using `testdatagen` to create data for test, pass `Stub: true` into the assertions when the object being created doesn't need to be saved to the database. Such as in instances when the test only needs to check for validation errors or when returning a mocked response. 

Reviewer: **suggestion:** Could we change  the second sentence ("Such as in instances") to "Examples of when you would want to use this are when the test only needs to check for validation errors or when returning a mocked response."

**praise:** Thanks for updating documentation!

**suggestion:** I don't know how much people monitor for PRs in this repo (I am not as diligent about it myself), but I'd recommend tagging people directly on the PR.

> - When using `testdatagen` to create data for test, pass `Stub: true` into the assertions when the object being created doesn't need to be saved to the database. Such as in instances when the test only needs to check for validation errors or when returning a mocked response. 

Reviewer 2: ```suggestion
- When using `testdatagen` to create data for test, pass `Stub: true` into the assertions when the object being created doesn't need to be saved to the database. Examples of when you would want to use this are when the test only needs to check for validation errors or when returning a mocked response. 
```
